tests/Parse/Source/Expression/Literal.elm

```diff
@@ -542,6 +556,167 @@ singleQuoteUnbalancedStringLiteralTest =
 
 
 
+-- TEMPLATE LITERALS -----------------------------------------------------------
+
+
+emptyTemplateLiteralTest : Test
+emptyTemplateLiteralTest =
+    let
+        title =
+            "Template literals may be empty."
+
+        input =
+            "``"
+
+        expected =
+            Ok <| Literal.Template []
+    in
+    test title
+        (\_ ->
+            Parser.run (Literal.parser Expression.local Expression.parser) input
+                |> Expect.equal expected
+        )
+
+
+textOnlyTemplateLiteralTest : Test
+textOnlyTemplateLiteralTest =
+    let
+        title =
+            "Template literals may contain only text."
+
+        input =
+            "`Hello foo`"
+
+        expected =
+            Ok <| Literal.Template [ Literal.Text "Hello foo" ]
+    in
+    test title
+        (\_ ->
+            Parser.run (Literal.parser Expression.local Expression.parser) input
+                |> Expect.equal expected
+        )
+
+
+textEscapeTemplateLiteralTest : Test
+textEscapeTemplateLiteralTest =
+    let
+        title =
+            "Template literals may contain string escapes."
+
+        input =
+            "`'Hello' \"\\`foo\\`\"\\n`"
+
+        expected =
+            Ok <|
+                Literal.Template
+                    [ Literal.Text "'Hello' \""
+                    , Literal.Text "`"
+                    , Literal.Text "foo"
+                    , Literal.Text "`"
+                    , Literal.Text "\""
+                    , Literal.Text "\n"
```

Comment:
You can see from this test that escapes bloat the list. I don't know if there's a good way to merge all consecutive Text elements of a list, or maybe append to the last element when parsing.

Reply:
Bear in mind that `"\""` is not a template escape, `"` just has to be escaped in the elm string.